### DISCUSSION TITLE
Can't search on 2.2.0 version of milvus - pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=index doesn't exist, collectionID 437775018607185420)>

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [milvus-io/milvus#20964] [Q&A and General discussion]
I create a collection with
```
fields = [
    FieldSchema(name="pk", dtype=DataType.INT64, is_primary=True, auto_id=False),
    FieldSchema(name="embeddings", dtype=DataType.FLOAT_VECTOR, dim=512)
]
schema = CollectionSchema(fields, "faces schema")
hello_milvus = Collection("faces", schema,  consistency_level="Eventually")
```

Then search by 

```
search_param = {
        "data": [res[0]['embeddings']],
        "anns_field": "embeddings",
        "param": {"metric_type": "L2", "params": {}},
        "limit": 10}
results = collection.search(**search_param)
```

But I get an error message  - Can't search on 2.2.0 version of milvus.

pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=index doesn't exist, collectionID 437775018607185420)>

**Comment by martinenkoEduard** [comment]
Have the syntax been changed on 2.2.0?
Or do I just need to upgrade on a newer version of pymilvus?

**Reply by yhmo** [reply]
In v2.x, we require user call **collection.load()** before **collection.search()**. In v2.2, we require user call **collection.create_index()** before **collection.load()**.
So, in v2.2, the correct steps are:
1. create a collection
2. call collection.create_index() to specify an index, if you want to do brute-force search, just create a FLAT index type.
3. insert data
4. call collection.load()
5. call collection.search()

You can swap the sequence of step 2 and step 3, it doesn't matter.
There is a good example: https://github.com/milvus-io/pymilvus/blob/2.2/examples/example.py

**Reply by martinenkoEduard** [reply]
What does collection.flush() do?

**Reply by xiaofan-luan** [reply]
> What does collection.flush() do?

Generally speak, you need to build index and load collection, pls check hellomilvus example.
Flush helps to sync all data to disk, you only need to do so before index build
